src/fetch/orderbook.py

```diff
@@ -0,0 +1,251 @@
+"""Basic client for connecting to postgres database with login credentials"""
+
+from __future__ import annotations
+
+import os
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+import pandas as pd
+from dotenv import load_dotenv
+from pandas import DataFrame
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+
+from src.logger import set_log
+from src.models.block_range import BlockRange
+from src.utils.query_file import open_query
+from src.dbt.common import node_suffix
+
+log = set_log(__name__)
+
+MAX_PROCESSING_DELAY = 10
+BUCKET_SIZE = {"mainnet": 10000, "xdai": 30000, "arbitrum-one": 1000000}
+
+
+class OrderbookEnv(Enum):
+    """
+    Enum for distinguishing between CoW Protocol's staging and production environment
+    """
+
+    BARN = "BARN"
+    PROD = "PROD"
+    ANALYTICS = "ANALYTICS"
+
+    def __str__(self) -> str:
+        return str(self.value)
+
+
+@dataclass
+class OrderbookFetcher:
+    """
+    A pair of Dataframes primarily intended to store query results
+    from production and staging orderbook databases
+    """
+
+    @staticmethod
+    def pg_engine(db_env: OrderbookEnv) -> Engine:
+        """Returns a connection to postgres database"""
+        load_dotenv()
+        if db_env == OrderbookEnv.ANALYTICS:
+            db_url = os.environ["ANALYTICS_DB_URL"]
+            db_string = f"postgresql+psycopg2://{db_url}"
+            return create_engine(db_string)
+
+        db_url = os.environ[f"{db_env}_DB_URL"]
+        db_string = f"postgresql+psycopg2://{db_url}"
+        return create_engine(db_string)
```

Comment:
```suggestion
        if db_env == OrderbookEnv.ANALYTICS:
            db_url = os.environ["ANALYTICS_DB_URL"]
        else:
            db_url = os.environ[f"{db_env}_DB_URL"]

        db_string = f"postgresql+psycopg2://{db_url}"
        return create_engine(db_string)
```
would be easier for me to parse